lib/package.rb

```diff
@@ -197,38 +191,30 @@ def self.depends_on(dependency, ver_range = nil)
     @dependencies.store(depName, [dep_tags, ver_check])
   end
 
-  def self.get_url(architecture)
-    if !@build_from_source && @binary_sha256 && @binary_sha256.key?(architecture)
-      return get_binary_url(architecture)
-    elsif @source_url.respond_to?(:has_key?)
-      return @source_url.key?(architecture) ? @source_url[architecture] : nil
+  def self.get_url
+    if is_binary?
+      return "https://gitlab.com/api/v4/projects/26210301/packages/generic/#{name}/#{version}_#{ARCH}/#{name}-#{version}-chromeos-#{ARCH}.#{binary_compression}"
+    elsif @source_url.is_a?(Hash)
+      return @source_url&.dig(ARCH.to_sym)
     else
       return @source_url
     end
   end
 
-  def self.get_binary_url(architecture)
-    architecture = 'armv7l' if architecture == 'aarch64'
-    return "https://gitlab.com/api/v4/projects/26210301/packages/generic/#{name}/#{version}_#{architecture}/#{name}-#{version}-chromeos-#{architecture}.#{binary_compression}"
-  end
-
-  def self.get_source_url(architecture) = @source_url.key?(architecture) ? @source_url[architecture] : nil
-
-  def self.get_sha256(architecture)
-    if !@build_from_source && @binary_sha256 && @binary_sha256.key?(architecture)
-      return @binary_sha256[architecture]
-    elsif @source_sha256.respond_to?(:has_key?)
-      return @source_sha256.key?(architecture) ? @source_sha256[architecture] : nil
+  def self.get_sha256
+    if is_binary?
+      return @binary_sha256[ARCH.to_sym]
+    elsif @source_sha256.is_a?(Hash)
+      return @source_sha256&.dig(ARCH.to_sym)
```

Comment:
Since we know it is a hash, we can assume it is not `nil` now (`.dig` is not necessary in this case):
```ruby
 return @source_sha256[ARCH.to_sym]
```

Reply:
That's true, but we don't know if it has the architecture we need, so we still need dig to return nil instead of [] which returns an error. We can drop the safe navigation tho